renderdoc/core/resource_manager.h

```diff
@@ -934,7 +957,7 @@ void ResourceManager<Configuration>::Serialise_InitialContentsNeeded(WriteSerial
   for(auto it = m_FrameReferencedResources.begin(); it != m_FrameReferencedResources.end(); ++it)
   {
     RecordType *record = GetResourceRecord(it->first);
-    if(IsDirtyFrameRef(it->second))
+    if(IsDirtyFrameRef(it->second) && !IsResourceSkipped(it->first))
```

Comment:
Why was this exclusion added here? This list is important because it contains the list of resources that are written to within the frame. Even for resources that don't have initial states (for example if they were created mid-frame then written to) they should be in this list. Then on replay any resource in this list which doesn't have serialised initial states will have dummy initial states created for it - usually just a marker saying the image needs to be cleared.

I'm not sure any resources could be in this list and still be skipped, because in `MarkResourceFrameReferenced` skipped resources are unconditionally removed from the skipped list if I understand right? If this is possible though then it would cause artifacts on replay - without the above created initial states then resources that are written won't be reset properly if you select an earlier event in the frame to replay  because they won't have anything to reset themselves to.

Admittedly the naming is extremely bad since this list has existed for a long time and its meaning has morphed over time (e.g. the DataInSerialiser is only used by D3D11 and is a legacy concept).

Reply:
FYI This wasn't removed in the latest push as far as I can see.

Reply:
Sorry I might have mistakenly undone the remove.